services/workflows-service/src/events/document-changed-webhook-caller.ts

```diff
@@ -3,23 +3,33 @@
 
 import { WorkflowEventRawData } from '@/workflow/workflow-event-emitter.service';
 import { Injectable } from '@nestjs/common';
-import { OnEvent, EventEmitter2 } from '@nestjs/event-emitter';
-import { ListenerFn } from './types';
-import axios from 'axios';
+import { EventEmitter2 } from '@nestjs/event-emitter';
 import { randomUUID } from 'crypto';
-import { sortBy } from 'lodash';
+import { HttpService } from '@nestjs/axios';
+import { ConfigService } from '@nestjs/config';
 
 @Injectable()
-export class EventConsumerListener {
-  constructor(private eventEmitter: EventEmitter2) {}
+export class DocumentChangedWebhookCaller {
+  #__axios: any;
+
+  constructor(
+    private httpService: HttpService,
+    private eventEmitter: EventEmitter2,
+    private configService: ConfigService,
+  ) {
+    this.#__axios = this.httpService.axiosRef;
+
+    eventEmitter.on('workflow.context.changed', this.handleWorkflowEvent.bind(this));
+  }
 
-  @OnEvent('workflow.context.changed')
```

Comment:
you are not using @OnEvent(...) eventually? i thought you found the way to use it earlier